pkg/controller/util.go

```diff
@@ -803,7 +807,7 @@ func getPriorityClass(pvc *v1.PersistentVolumeClaim) string {
 }
 
 func shouldDeletePod(pvc *v1.PersistentVolumeClaim) bool {
-	return pvc.GetAnnotations()[AnnPodRetainAfterCompletion] != "true" || pvc.DeletionTimestamp != nil
+	return pvc.GetAnnotations()[AnnPodRetainAfterCompletion] != "true" || pvc.GetAnnotations()[AnnRequiresScratch] == "true" || pvc.DeletionTimestamp != nil
```

Comment:
so we need to delete if requires scratch?

Reply:
Yeah so the mechanism used to determine if we need scratch is:
1. If known we need scratch just create a pod with scratch.
2. If unknown, we start a pod without scratch, if during that pods run it determines we need scratch it exits with a special exit code indicating we need scratch space.
  a. The import controller reads the exit code and sets an annotation on the PVC indicating we need scratch space.
  b. Next we delete the import pod.
  c. Next reconcile loop, we see the scratch space annotation, and create a new pod with scratch space.

Now without this PR, if you also set the don't delete pod annotation, then step b will fail, and no new pod is created because the old pod still exists.